charts/templates/secret.yaml

```diff
@@ -1,3 +1,9 @@
+{{- $adminPassword := (randAlphaNum 8) | b64enc | quote }}
+{{- $secret := (lookup "v1" "Secret" .Release.Namespace .Release.Name ) }}
```

Comment:
Do we need this code block here, can you please provide some explanation?

Reply:
This block looks up if the secret already exists or not, so during an upgrade of the chart it will use the existing secret and not generate a new random password.